src/ast.rs

```diff
@@ -5,18 +5,24 @@ pub struct Call {
 }
 
 #[derive(Debug)]
-pub enum UnaryPlus {
-    Integer(i32),
-    FloatingPoint(f64),
-    Identifier(String),
+pub enum PlusOrMinus {
+    Plus,
+    Minus,
 }
+
 #[derive(Debug)]
-pub enum UnaryMinus {
+pub enum UnaryTarget {
     Integer(i32),
-    FloatingPoint(f64),
+    Float(f64),
     Identifier(String),
 }
 
+#[derive(Debug)]
+pub struct UnaryOp {
+    pub sign: PlusOrMinus,
```

Comment:
Do we want to be able to generalise this to all unary operators (e.g. logical negation `!`, bitwise inversion `~`, ..)?
If so, we may want to name this field according to the general case -- `operation` rather than `sign`

Reply:
addressed